src/__experimental__/components/numeral-date/numeral-date.component.js

```diff
@@ -15,18 +17,32 @@ const NumeralDate = ({
   name,
   onBlur,
   onChange,
-  value
+  value,
+  validationOnLabel = false,
+  label,
+  labelInline,
+  labelWidth,
+  labelAlign,
+  labelHelp,
+  fieldHelp
 }) => {
+  const { current: uniqueId } = useRef(id || guid());
   const isControlled = value !== undefined;
   const initialValue = isControlled ? value : defaultValue;
-  invariant(
```

Comment:
It might be worth having an invariant to cover switching between controlled and uncontrolled

Reply:
Why did we remove this invariant?

Reply:
It was not possible to have this component work without any value provided.